src/explore-education-statistics-common/src/modules/table-tool/utils/createExpandedRowHeaders.ts

```diff
@@ -91,31 +91,33 @@ function getRowHeadersInfo(rowHeaders: Header[]): {
 
   rowHeaders.forEach(header => {
     const { maxCrossSpan } = header;
+    const stack = [header];
 
-    let current: Header | undefined = header;
+    while (stack.length > 0) {
+      const current = stack.shift();
+
+      if (!current) {
+        break;
+      }
 
-    while (current) {
       const { depth } = current;
 
-      if (current.parent) {
+      if (current.parent && collapsibleLevels[depth] !== false) {
         const isCollapsibleRowHeaderLevel =
           current.hasSiblings() &&
           current.parent.children.every(child =>
             child.hasSingleMatchingChild(),
           );
 
-        collapsibleLevels[depth] =
-          (collapsibleLevels[depth] ?? true) && isCollapsibleRowHeaderLevel;
+        collapsibleLevels[depth] = isCollapsibleRowHeaderLevel;
       }
 
-      current = current.getFirstChild();
+      if (current.hasChildren()) {
```

Comment:
If we do the above optimisation around `!current.hasChildren`, we don't need this condition anymore